src/conways-game-of-life/index.ts

```diff
@@ -0,0 +1,71 @@
+const getGridSize = (grid: number[][]) => {
+    const colMax = grid[0].length-1;
+    const rowMax = grid.length-1;
+    return { rowMax, colMax }
+}
+
+const updateCellStatus = (currentCellStatus: number, neighbourSum: number) => {
+    if (currentCellStatus === 1 && neighbourSum > 3) {
+        return 0;
+    }
+    else if (currentCellStatus === 1 && neighbourSum < 2) {
+        return 0;
+    }
+    else if (currentCellStatus === 0 && neighbourSum === 3) {
+        return 1;
+    }
+    return currentCellStatus;
+} //Apply the conditions for updating the status of a cell 
+
+const calculateNeighbourSum = (grid: number[][], rowIndex: number, colIndex: number) => {
+    let neighbourSum = 0;
+    const gridSize = getGridSize(grid);
+    for (let j = Math.max(rowIndex - 1, 0); j <= Math.min(rowIndex + 1, gridSize.rowMax); j++) {
+        for (let i = Math.max(colIndex - 1, 0); i <= Math.min(colIndex + 1, gridSize.colMax); i++) {
+            if (i === colIndex && j === rowIndex) continue;
+            neighbourSum += grid[j][i];
+        }
+    }
```

Comment:
That is amazing. You got it, good job. The only thing missing (or extra depending how you look at it) is the getGridSize function. It's seems unnecessary. You can easily get the size based on i and j length.

If I can be pick, usually in a nested loop i is the outer and j is the inner loop - that is me been pick, you don't need to change that.

Reply:
Thanks!